Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -54,18 +56,29 @@ public class PersistentConnection : IPersistentConnection
             this.eventBus = eventBus;
         }
 
-        /// <inheritdoc />
-        public IModel CreateModel()
+        public void Connect()
+        {
+            initializedConnection = Initialize();
+        }
+
+        private IAutorecoveringConnection Initialize()
         {
+
             var connection = initializedConnection;
             if (connection == null)
                 lock (mutex)
-                    connection = initializedConnection ??= Connect();
+                    connection = initializedConnection ??= ConnectInternal();
```

Comment:
ConnectInternal() creates the connection and fires the connected event however IsConnected is still false at the time since it will be set when ConnectInternal() returns or am I missing something?

Reply:
to me it works as expected......finally 😄 

![image](https://user-images.githubusercontent.com/1633595/133793929-11f600a3-d4cf-4f2a-a26e-9bb901a3f705.png)

Reply:
and if I don't explicitly connect it hung, again as expected

![image](https://user-images.githubusercontent.com/1633595/133794119-1b21f1f9-e054-4f4b-9c3d-f9cb3e6634f2.png)

but if you have at least one subscriber or some other commands going on in your app/service (this is how my 99% of services work), then it auto-connect and works again as expected

![image](https://user-images.githubusercontent.com/1633595/133794427-0ce5dac7-7c92-4ba6-bdf1-235dfe2ab29e.png)